src/components/DrawingBoard.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { MouseEvent, useEffect, useRef } from 'react';
+import './DrawingBoard.css';
+
+export interface DrawingBoardProps {
+    imageUrl: string;
+    index: number;
+}
+
+enum drawingState {
+    NotStarted,
+    StartPoint,
+    Completed,
+}
+
+interface DrawingLine {
+    x1: number,
+    y1: number,
+    x2: number,
+    y2: number,
+}
+
+
+const DrawingBoard: React.FC<DrawingBoardProps> = (props: DrawingBoardProps) => {
+
+    let history: DrawingLine[] = [];
+
+    // save the reference to the canvas html element
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+
+    let currentDrawing = {
+        state: drawingState.NotStarted,
+        x1: 0,
+        y1: 0,
+        x2: 0,
+        y2: 0,
+
+    }
+
+    const onCanvasClick = (event: MouseEvent<HTMLCanvasElement>) => {
+
+        const rect = (canvasRef.current as unknown as HTMLCanvasElement).getBoundingClientRect();
+        const x = event.clientX - rect.left;
+        const y = event.clientY - rect.top;
+
+        if (currentDrawing.state === drawingState.NotStarted) {
+
+            currentDrawing.x1 = x;
+            currentDrawing.y1 = y;
+            currentDrawing.state = drawingState.StartPoint;
+        }
+
+        else if (currentDrawing.state === drawingState.StartPoint) {
+            currentDrawing.x2 = x;
+            currentDrawing.y2 = y;
+            currentDrawing.state = drawingState.Completed;
+        }
+
+
+        if (currentDrawing.state === drawingState.Completed) {
+
+            let line: DrawingLine = {
+                x1: currentDrawing.x1,
+                y1: currentDrawing.y1, 
+                x2: currentDrawing.x2, 
+                y2: currentDrawing.y2
+            };
+
+            // draw line on the canvas
+            drawHistoryLine(line);
+
+            // push to the history
+            history.push(line);
+
+            // save in local storage
+            localStorage[props.index] = JSON.stringify(history);
+
+            // clear current drawing state
+            currentDrawing = {
+                state: drawingState.NotStarted,
+                x1: 0,
+                y1: 0,
+                x2: 0,
+                y2: 0,
+
+            }
+        }
+    };
+
+
+    const onClearClick = () => {
+
+        //clear output
+        console.log("Clear clicked");
+        history = [];
+
+
+        // clear history also on local storage
+        localStorage[props.index] = JSON.stringify(history);
+
+        drawImage(props.imageUrl);
+    };
+
+
+    async function drawImage(url: string) {
```

Comment:
why is this async?
in addition, in the uses I saw, you treat it as a sync function